### ISSUE TITLE
Create site JSON

### ISSUE THREAD

**Issue by KevinWorkman** [KevinWorkman/HappyCoding#6]
This mostly just sounds fun to do. Create a json file (through Jekyll magic) that contains info about every tutorial and example.

Then use that file to create a random link to a random tutorial, with a teaser. Something like this:

Want to learn how to make a Random Walker? Click here!

This would also involve adding the teaser info to the frontmatter.

**Comment by bigbass1997** [comment]
Could you try explaining this idea more? I'm having trouble understanding what such a JSON file would be used for and how it would be accessed.

**Comment by KevinWorkman** [comment]
The general idea is to have a little "advertisement" at the top or bottom of a page. Something like: Want to learn XYZ? Click here!

The XYZ part should be randomly chosen from all of the tutorials. The problem is where to get that XYZ from. 

The site uses Jekyll (and a language called Liquid) which can programatically create pages: the tutorial listing is created from a loop that iterates over all the tutorials and puts together the html, for example.

It would be neat to use Jekyll to put together a JSON file that is then read by JavaScript (using ajax) and chooses a random tutorial from that JSON to create that random link.

This requires understanding how Jekyll and Liquid works, but I think it's doable. Then from there that JSON file might be used for other stuff (like a miniature api), but that's down the road.

**Comment by KevinWorkman** [comment]
I've added a super basic version of what I was talking about. Turned out to be easier than I expected.

View the source here: https://github.com/KevinWorkman/HappyCoding/blob/gh-pages/site.json

Which results in this file: http://happycoding.io/site.json

More info on how it works here: http://jekyll.tips/jekyll-casts/output-json/

I might move this file, or maybe split it into separate files for tutorials and examples, and it definitely needs more info (such as the "teaser" I mentioned). But hopefully this shows what I meant. Opens up a couple cool options.
